src/main/java/org/opensearch/agent/tools/SearchMonitorsTool.java

```diff
@@ -0,0 +1,226 @@
+        this.client = client;
+
+        // probably keep this overridden output parser. need to ensure the output matches what's expected
+        outputParser = new Parser<>() {
+            @Override
+            public Object parse(Object o) {
+                @SuppressWarnings("unchecked")
+                List<ModelTensors> mlModelOutputs = (List<ModelTensors>) o;
+                return mlModelOutputs.get(0).getMlModelTensors().get(0).getDataAsMap().get("response");
+            }
+        };
+    }
+
+    // Response is currently in a simple string format including the list of monitors (only name and ID attached), and
+    // number of total monitors. The output will likely need to be updated, standardized, and include more fields in the
+    // future to cover a sufficient amount of potential questions the agent will need to handle.
+    @Override
+    public <T> void run(Map<String, String> parameters, ActionListener<T> listener) {
+        final String monitorId = parameters.getOrDefault("monitorId", null);
+        final String monitorName = parameters.getOrDefault("monitorName", null);
+        final String monitorNamePattern = parameters.getOrDefault("monitorNamePattern", null);
+        final Boolean enabled = parameters.containsKey("enabled") ? Boolean.parseBoolean(parameters.get("enabled")) : null;
+        final Boolean hasTriggers = parameters.containsKey("hasTriggers") ? Boolean.parseBoolean(parameters.get("hasTriggers")) : null;
+        final String indices = parameters.getOrDefault("indices", null);
+        final String sortOrderStr = parameters.getOrDefault("sortOrder", "asc");
+        final SortOrder sortOrder = sortOrderStr == "asc" ? SortOrder.ASC : SortOrder.DESC;
+        final String sortString = parameters.getOrDefault("sortString", "monitor.name.keyword");
+        final int size = parameters.containsKey("size") ? Integer.parseInt(parameters.get("size")) : 20;
+        final int startIndex = parameters.containsKey("startIndex") ? Integer.parseInt(parameters.get("startIndex")) : 0;
+
+        // If a monitor ID is specified, all other params will be ignored. Simply return the monitor details based on that ID
+        // via the get monitor transport action
+        if (monitorId != null) {
+            GetMonitorRequest getMonitorRequest = new GetMonitorRequest(monitorId, 1L, RestRequest.Method.GET, null);
+            ActionListener<GetMonitorResponse> getMonitorListener = ActionListener.<GetMonitorResponse>wrap(response -> {
+                StringBuilder sb = new StringBuilder();
+                Monitor monitor = response.getMonitor();
+                if (monitor != null) {
+                    sb.append("Monitors=[");
+                    sb.append("{");
+                    sb.append("id=").append(monitor.getId()).append(",");
+                    sb.append("name=").append(monitor.getName());
+                    sb.append("}]");
+                    sb.append("TotalMonitors=1");
```

Comment:
will that be `1` all the time? What if we find 2/3 monitors?

Reply:
This will be `1` all of the time since it is a `GetMonitorResponse`: https://opensearch.org/docs/latest/observing-your-data/alerting/api/#get-monitor

Reply:
See comment on line 93/94 above as well.